src/envoy/tools.clj

```diff
@@ -64,10 +64,11 @@
   the numbers to longs, the alphanumeric values to strings, and will use Clojure reader for the rest
   in case reader can't read OR it reads a symbol, the value will be returned as is (a string)"
   (cond
-    (re-matches #"[0-9]+" v) (Long/parseLong v)
+    (nil? v)                         v
```

Comment:
you sure it won't be false positive for:
* _paths_ that are missing
OR
* having an empty string as a value (that could come as "`nil`")?

Reply:
@tolitius It won't be false positive, because  when we set value in CONSUL to explicitly set as `nil` we specify the same which will be converted to `String` and also will only replace as `nil` for explicit mentioning